File: AndroidStudioProjects/yogainstructorapp/app/src/main/java/com/example/yogainstructorapp/repository/UserRepository.java
package com.example.yogainstructorapp.repository;

import com.example.yogainstructorapp.data.UserDao;

public class UserRepository {
    private final UserDao userDao;

    public UserRepository(UserDao userDao) {
        this.userDao = userDao;
    }

    // Add your data manipulation methods here
}

File: AndroidStudioProjects/yogatry1/app/UserRepository.kt
package com.example.yogainstructorapp.repository

import com.example.yogainstructorapp.data.User
import com.example.yogainstructorapp.data.UserDao

class UserRepository(private val userDao: UserDao) {
    suspend fun registerUser(username: String, password: String) {
        val user = User(username = username, password = password)
        userDao.insert(user)
    }

    suspend fun loginUser(username: String, password: String): Boolean {
        return userDao.getUser(username, password) != null
    }
}
